nikola/plugins/command/auto/__init__.py

```diff
@@ -0,0 +1,302 @@
+
+        self.command_line = 'nikola ' + ' '.join(arguments)
+
+        # Run an initial build so we are up-to-date
+        subprocess.call(["nikola"] + arguments)
+
+        port = options and options.get('port')
+        self.snippet = '''<script>document.write('<script src="http://'
+            + (location.host || 'localhost').split(':')[0]
+            + ':{0}/livereload.js?snipver=1"></'
+            + 'script>')</script>
+        </head>'''.format(port)
+
+        watched = [
+            self.site.configuration_filename,
+            'themes/',
+            'templates/',
+        ]
+        for item in self.site.config['post_pages']:
+            watched.append(os.path.dirname(item[0]))
+        for item in self.site.config['FILES_FOLDERS']:
+            watched.append(item)
+        for item in self.site.config['GALLERY_FOLDERS']:
+            watched.append(item)
+        for item in self.site.config['LISTINGS_FOLDERS']:
+            watched.append(item)
+
+        out_folder = self.site.config['OUTPUT_FOLDER']
+        if options and options.get('browser'):
+            browser = True
+        else:
+            browser = False
+
+        if options['ipv6']:
+            dhost = '::'
+        else:
+            dhost = None
+
+        host = options['address'].strip('[').strip(']') or dhost
+
+        # Start watchers that trigger reloads
+        reload_wm = pyinotify.WatchManager()
+        reload_notifier = pyinotify.ThreadedNotifier(reload_wm, self.do_refresh)
+        reload_notifier.start()
+        reload_wm.add_watch(out_folder, MASK, rec=True)  # Watch output folders
+
+        # Start watchers that trigger rebuilds
+        rebuild_wm = pyinotify.WatchManager()
+        rebuild_notifier = pyinotify.ThreadedNotifier(rebuild_wm, self.do_rebuild)
+        rebuild_notifier.start()
+        for p in watched:
+            if os.path.exists(p):
+                rebuild_wm.add_watch(p, MASK, rec=True)  # Watch input folders
+
+        parent = self
+
+        class Mixed(WebSocketWSGIApplication):
+            """A class that supports WS and HTTP protocols in the same port."""
+            def __call__(self, environ, start_response):
+                if environ.get('HTTP_UPGRADE') is None:
+                    return parent.serve_static(environ, start_response)
+                return super(Mixed, self).__call__(environ, start_response)
+
+        ws = make_server(
+            host, port, server_class=WSGIServer,
+            handler_class=WebSocketWSGIRequestHandler,
+            app=Mixed(handler_cls=LRSocket)
+        )
+        ws.initialize_websockets_manager()
+        self.logger.info("Serving on port {0}...".format(port))
+
+        # Yes, this is racy
+        if browser:
+            webbrowser.open('http://{0}:{1}'.format(host, port))
+
+        try:
+            ws.serve_forever()
+        except KeyboardInterrupt:
+            ws.server_close()
```

Comment:
Log this [like in the serve](https://github.com/getnikola/nikola/blob/master/nikola/plugins/command/serve.py#L123) plugin.